src/wordrec/language_model.cpp

```diff
@@ -1375,7 +1375,12 @@ void LanguageModel::ExtractFeaturesFromPath(const ViterbiStateEntry &vse, float
   // features[PTRAIN_NUM_BAD_FONT] = vse.consistency_info.inconsistent_font;
 
   // Classifier-related features.
-  features[PTRAIN_RATING_PER_CHAR] = vse.ratings_sum / static_cast<float>(vse.outline_length);
+  if (vse.outline_length != 0.0f) {
```

Comment:
```c
features[PTRAIN_RATING_PER_CHAR] = (vse.outline_length > 0.0f) ? (vse.ratings_sum / static_cast<float>(vse.outline_length)) : 0.0f;
```

Reply:
Why would that be better?

Reply:
Hi @stweil .

:warning: `(vse.outline_length != 0.0f)`!!!

Reply:
Thanks. I replaced `!=` by `>` now. Please review the updated code.

Reply:
Hi @stweil .

`vse.outline_length` it's `(float)`?